docs/sources/auth/generic-oauth.md

```diff
@@ -64,17 +66,21 @@ Grafana will attempt to determine the user's e-mail address by querying the OAut
 1. Query the `/emails` endpoint of the OAuth provider's API (configured with `api_url`) and check for the presence of an e-mail address marked as a primary address.
 1. If no e-mail address is found in steps (1-4), then the e-mail address of the user is set to the empty string.
 
-Grafana will also attempt to do role mapping through OAuth as described below.
+### Roles
+
+Grafana will check for the presence of a role using the [JMESPath](http://jmespath.org/examples.html) specified via the `role_attribute_path` configuration option. The JMESPath will be applied to the `id_token` first. If there is no match, the UserInfo endpoint specified via the `api_url` configuration option will be tried next. The result after evaluating the `role_attribute_path` JMESPath expression needs to be a valid Grafana role, i.e. `Viewer`, `Editor` or `Admin`.
 
-Check for the presence of a role using the [JMESPath](http://jmespath.org/examples.html) specified via the `role_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option. The result after evaluating the `role_attribute_path` JMESPath expression needs to be a valid Grafana role, i.e. `Viewer`, `Editor` or `Admin`.
+See [JMESPath examples](#jmespath-examples) for more information.
 
-Grafana also attempts to map teams through OAuth as described below.
+### Groups / Teams
 
-Check for the presence of groups using the [JMESPath](http://jmespath.org/examples.html) specified via the `groups_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option. After evaluating the `groups_attribute_path` JMESPath expression, the result should be a string array of groups.
+Similarly, group mappings can be made using [JMESPath](http://jmespath.org/examples.html) with the `groups_attribute_path` configuration option. The `id_token` will be attempted first, followed by the UserInfo from the `api_url`. The result of the JMESPath expression should be a string array of groups.
```

Comment:
```suggestion
Similarly, group mappings are made using [JMESPath](http://jmespath.org/examples.html) with the `groups_attribute_path` configuration option. The `id_token` is attempted first, followed by the UserInfo from the `api_url`. The result of the JMESPath expression should be a string array of groups.
```

Reply:
As above, think documentation tense change should be separate from this PR.